File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.java
class Solution {
    public int numSteps(String s) {
        int ans=0;
        int i=s.length()-1;
        while (s.charAt(i)=='0'){
            ans++;
            i--;
        }
        if(i==0) return ans;
        while (i>=0){
            if (s.charAt(i)=='1'){
                ans++;
                i--;
                continue;
            }
            while (s.charAt(i)=='0'){
                ans+=2;
                i--;
            }
        }
        return ans+1;

    }
}

File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        ans=0
        i = len(s)-1
        while s[i] =="0":
            ans+=1
            i-=1
        if i==0:
            return ans
        while i>=0:
            if s[i] == "1":
                ans += 1
                i-=1
                continue
            while s[i]=="0":
                ans+=2
                i-=1
        return ans+1
